packages/forms/src/directives/validators.ts

```diff
@@ -177,20 +182,23 @@ export const MAX_VALIDATOR: StaticProvider = {
   selector:
       'input[type=number][max][formControlName],input[type=number][max][formControl],input[type=number][max][ngModel]',
   providers: [MAX_VALIDATOR],
-  host: {'[attr.max]': 'max ?? null'}
+  host: {'[attr.max]': 'enabled() ? max : null'}
 })
 export class MaxValidator extends AbstractValidatorDirective implements OnChanges {
   /**
    * @description
    * Tracks changes to the max bound to this directive.
    */
-  @Input() max!: string|number;
+  @Input() max!: string|number|null;
   /** @internal */
   override inputName = 'max';
   /** @internal */
   override normalizeInput = (input: string): number => parseFloat(input);
```

Comment:
The `number` type is missing here (it's pre-existing, but we can improve this in this PR as well):
```suggestion
  override normalizeInput = (input: string|number): number => parseFloat(input);
```